frontend/src/lib/components/InsightLegend/InsightLegend.tsx

```diff
@@ -0,0 +1,69 @@
+import './InsightLegend.scss'
+import React from 'react'
+import { Button, Row, Col } from 'antd'
+import clsx from 'clsx'
+import { useActions, useValues } from 'kea'
+import { LegendIcon } from 'lib/components/icons'
+import { insightLogic } from 'scenes/insights/insightLogic'
+import { trendsLogic } from 'scenes/trends/trendsLogic'
+import { InsightLabel } from 'lib/components/InsightLabel'
+import { getChartColors } from 'lib/colors'
+import { PHCheckbox } from 'lib/components/PHCheckbox'
+import { formatCompareLabel } from 'scenes/insights/InsightsTable/InsightsTable'
+
+export function InsightLegendButton(): JSX.Element {
+    const { filters } = useValues(insightLogic)
+    const { toggleInsightLegend } = useActions(insightLogic)
+
+    return (
+        <Button className="insight-legend-button" onClick={toggleInsightLegend}>
+            <LegendIcon />
+            <span className="insight-legend-button-title">{filters.legend_hidden ? 'Show' : 'Hide'} Legend</span>
+        </Button>
+    )
+}
+
+export function InsightLegend(): JSX.Element {
+    const { insightProps, filters } = useValues(insightLogic)
+    const logic = trendsLogic(insightProps)
+    const { indexedResults, visibilityMap } = useValues(logic)
+    const { toggleVisibility } = useActions(logic)
+    const colorList = getChartColors('white', indexedResults.length, !!filters.compare)
+
+    return (
+        <div className={clsx('insight-legend-menu', indexedResults.length <= 5 && 'short')}>
+            {indexedResults &&
+                indexedResults.map((item) => {
```

Comment:
I'm not sure why this is happening, but the legend has a different order than the table below. You can see this if you turn on compare previous. The order here seems better, but either way they should probably be consistent.

Reply:
I think I figured it out. The table comes presorted by total sum column and canceling this returns the table to it's original correct order. I've changed it so that we don't sort by total sum by default.